src/silly_sprites/opengl/buffer_object.cpp

```diff
@@ -0,0 +1,97 @@
+#include "buffer_object.hpp"
+
+namespace sly::gl {
+    void BufferObject::init() {
+        generate_vertex_array();
+        generate_vertex_buffer();
+        generate_element_buffer();
+
+        glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), static_cast<void*>(0));
+        glEnableVertexAttribArray(0);
+    }
+
+    void BufferObject::generate_vertex_array() {
+        glGenVertexArrays(1, &m_vao);
+        bind_vertex_array();
+    }
+
+    void BufferObject::generate_vertex_buffer() {
+        glGenBuffers(1, &m_vbo);
+        bind_vertex_buffer();
+        data_vertex_buffer();
+    }
+
+    void BufferObject::generate_element_buffer() {
+        glGenBuffers(1, &m_ebo);
+        bind_element_buffer();
+        data_element_buffer();
+    }
+
+    void BufferObject::bind_vertex_array() const {
+        glBindVertexArray(m_vao);
+    }
+
+    void BufferObject::bind_vertex_buffer() const {
+        glBindBuffer(GL_ARRAY_BUFFER, m_vbo);
+    }
+
+    void BufferObject::bind_element_buffer() const {
+        glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, m_ebo);
+    }
+
+    void BufferObject::data_vertex_buffer() const {
+        glBufferData(GL_ARRAY_BUFFER, static_cast<GLsizeiptr>(sizeof(float) * m_points.size()), m_points.data(), GL_STATIC_DRAW);
+        /*
+            ugly cast because of ubuntu.
+            you can see the error here: https://github.com/mgerhold/silly-sprites/actions/runs/5992560779/job/16252072736
+            maybe you have a better idea.
+        */
+    }
+
+    void BufferObject::data_element_buffer() const {
+        glBufferData(GL_ELEMENT_ARRAY_BUFFER, static_cast<GLsizeiptr>(sizeof(unsigned int) * m_indices.size()), m_indices.data(), GL_STATIC_DRAW);
+        // ugly cast again
+    }
+
+    BufferObject::BufferObject(Points points, Indices indices) : m_points{ points }, m_indices{ indices } {
+        init();
+    }
+
+    BufferObject::~BufferObject() {
+        glDeleteBuffers(1, &m_vbo);
+        glDeleteBuffers(1, &m_ebo);
+        glDeleteVertexArrays(1, &m_vao);
```

Comment:
I think the order of destruction should be the inverse of the order of construction (which it isn't here). You *could* (but don't have to) write wrappers for each of those buffer types. If you make those buffer types members of this class, then the order of destruction would be automatically be the inverse of construction. In my opinion this could also improve readability of the code in this file.

But please decide for yourself if you want to do that.